### PR TITLE
Google Calendar Update

### LINKED ISSUE #245: Tutor Schedules -> Google Calendar
Kristen (one of the writing center admins) said it would be nice if the tutors could have a way to put their schedules into their google calendars automatically. This seems like something cool worth looking into.

### PR DESCRIPTION
**Pull request by bam95899**
## Description

This PR will add the ability to add your writing center appointments/shifts to your Google Calendar.

After logging into your Bethel Google Account by clicking on a "login to google calendar" button, you can then either click a button to add all upcoming events to Google Calendar, or click on a specific event and add that specific event to your Calendar. Clicking both buttons won't create duplicate events.

Adding appointments to your Google Calendar is supported if you are a student or tutor.

Fixes #245 

## Size and Type of change

- Small Change - 1 person needs to review this

- New feature

- Venv update needed
    - Install new requirements

- Config update needed
    - Add `GC_CLIENT_SECRET` location to config

- Add `gc_client_secret.json` file to project

## How Has This Been Tested?

Locally I implemented the changes and tested adding appointments to my Google Calendar using both buttons (add specific event/add all upcoming events)

I then moved the code over to xp and did the same tests and they succeeded.

## Checklist:

Only check what applies and what you have done.

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have tested on multiple browsers (Chrome, Firefox, Safari, IE suite)

### CODE REVIEW

File: writing_center/google_calendar/__init__.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+        service = build('calendar', 'v3', credentials=credentials, cache_discovery=False)
+
+        user = self.gcc.get_user_by_username(flask_session['USERNAME'])
+
+        page_type = str(json.loads(request.data).get('page_type'))
+
+        self.gcc.handle_events(user, page_type, service)
+
+        return ''
```
Comment (c-schwarze) [inline writing_center/google_calendar/__init__.py]: Probably worth returning something - even if it is `return True`

File: writing_center/templates/macros/add-events-to-calendar.html [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                }),
+                contentType: 'application/json;charset=UTF-8',
+                success: function (response) {
+                    window.location.replace(response);
+                },
+                error: function (error) {
+
+                }
+            });
+        });
```
Comment (c-schwarze) [inline writing_center/templates/macros/add-events-to-calendar.html]: This code is duplicated in writing_center/templates/macros/appointment_modal.html. Can we reuse that?

Comment (bam95899) [reply writing_center/templates/macros/add-events-to-calendar.html]: Made a macro for login

File: writing_center/google_calendar/__init__.py (hunk 1 above)
Comment (bam95899) [reply writing_center/google_calendar/__init__.py]: returned 'success'

**Pull request merged**